mlflow/tracing/fluent.py

```diff
@@ -549,3 +550,133 @@ def f():
         return TRACE_BUFFER.get(last_active_request_id)
     else:
         return None
+
+
+@experimental
+def use_remote_trace(trace: Trace):
+    """
+    Merge a trace from a remote service into the current active trace.
+    If there is an active span, the remote trace will be merged under that span. Otherwise,
+    a new trace is created locally that copies all spans and metadata from the remote trace.
+
+    The following example demonstrates how to use this function to merge a trace from a remote
+    service to the current active trace.
+
+    ..code-block:: python
+
+        @mlflow.trace(name="apple")
+        def run():
+            # Call a remote service that returns a trace in the response
+            resp = requests.get("https://your-service-endpoint", ...)
+
+            # Extract the trace from the response
+            trace_json = resp.json().get("trace")
+            trace = Trace.from_dict(trace_json)
+
+            # Use the remote trace as a part of the current active trace.
+            # It will be merged under the span "apple" and exported together when it is ended.
+            mlflow.use_remote_trace(trace)
+
+    Args:
+        trace: A :py:class:`Trace <mlflow.entities.Trace>` object that represents the remote trace.
+            The trace **must** be already completed i.e. no further updates should be made to it.
+            Otherwise, this function will raise an exception.
+
+    Raises:
+        MlflowException: If there is no active span found.
+    """
+    if not isinstance(trace, Trace):
+        raise MlflowException.invalid_parameter_value(
+            f"Invalid trace object: {type(trace)}. Please provide a valid MLflow Trace object "
+            "to use it as a remote trace. You can create a Trace object from its json format by "
+            "using the Trace.from_dict() method."
+        )
+
+    if trace.info.status not in TraceStatus.end_statuses():
+        raise MlflowException.invalid_parameter_value("The remote trace must be ended.")
+
+    if active_span := get_current_active_span():
+        # If there is an active span, merge the new trace under the active one
+        _merge_remote_trace_to_local(
+            remote_trace=trace,
+            local_request_id=active_span.request_id,
+            local_parent_span_id=active_span.span_id,
+        )
+    else:
+        # If there is no active span, create a new root span named "Remote Trace <...>"
+        # and put the remote trace under it. This design aims to keep the trace export
+        # logic simpler and consistent, rather than directly exporting the remote trace.
+        client = MlflowClient()
+        remote_root_span = trace.data.spans[0]
+        span = client.start_trace(
```

Comment:
This effectively merges a single span. What if the remote response JSON contains multiple spans with parent/child relationships? Is this the motivation behind forcing users to submit a `Trace` object? 
While this is fine for a correctness approach, we may want to later think (if users complain about the complexity of span parsing from remote provider systems and the multiple calls to this API to reason about the hierarchical relationship preservation) about visiting this topic. YAGNI for now, though